crates/sui-core/src/transaction_input_checker.rs

```diff
@@ -251,7 +256,9 @@ async fn check_objects(
     // If any errors with the locks were detected, we return all errors to give the client
     // a chance to update the authority if possible.
     if !errors.is_empty() {
-        return Err(SuiError::TransactionInputObjectsErrors { errors });
+        return Err(SuiError::TransactionInputObjectsErrors {
```

Comment:
Not sure I understand the rationale for double wrapping here could you elaborate?

Reply:
by `double wrapping`, do you mean why we use a `TransactionInputObjectsErrors` to wrap other various errors? 
If so, it's because grouping errors helps with categorization errors. For example, we consider input error as a user error, this non-retryable

Reply:
I get why we're doing it in general, but here were wrapping it in two layers of the same error type, so that's what this comment is about.

Reply:
can you point where we could see a `TransactionInputObjectsErrors {TransactionInputObjectsErrors}` if this is what you mean? btw, I just updated the PR to make it cleaner and address your comments

Reply:
In the first commit, where this thread is rooted, the line that I put it on is an example of a `TransactionInputObjectsErrors { TransactionInputObjectsErrors }` but it is gone in the second commit, so there's no example anymor!

Reply:
cool, thanks!